src/components/ApiAction/ImportActionsModal.vue

```diff
@@ -0,0 +1,192 @@
+                  size="sm"
+                  :data-cy="`imported-action-name-editor-${data.index}`"
+                  :state="!savedActionNames.includes(data.item.name)"
+                  v-model="data.item.name"
+                >
+                </b-form-input>
+                <b-form-invalid-feedback
+                  :id="`name-input-${data.index}-feedback`"
+                >
+                  You already have a saved action with that name.
+                </b-form-invalid-feedback>
+              </template>
+            </b-table>
+          </b-card>
+        </b-col>
+      </b-row>
+    </b-container>
+  </b-modal>
+</template>
+
+<script>
+/* eslint-disable */
+export default {
+  name: 'import-actions-modal',
+  props: {
+    savedActionNames: {
+      required: true
+    }
+  },
+  data() {
+    return {
+      file: null,
+      files: [],
+      actions: [],
+      tableFields: [
+        { key: 'name', label: 'Name' },
+        { key: 'query.controller', label: 'Controller' },
+        { key: 'query.action', label: 'Action' }
+      ]
+    }
+  },
+  computed: {
+    validNames() {
+      return this.actions.some(action =>
+        this.savedActionNames.includes(action.name)
+      )
+    }
+  },
+  watch: {
+    file(value) {
+      if (value) {
+        this.fileAdded(value)
+      }
+    },
+    files(value) {
+      this.filesChange(value)
+    }
+  },
+  methods: {
+    resetForm() {
+      this.file = null
+      this.files = []
+      this.actions = []
+    },
+    handleImport() {
+      this.$emit('import-actions', this.actions)
+    },
+    async isInvalidFile(file) {
+      if (this.files.find(f => f.name === file.name && f.size === file.size)) {
+        this.$bvToast.toast(`File ${this.file.name} already added!`, {
+          title: 'Warning',
+          variant: 'warning',
+          toaster: 'b-toaster-bottom-right',
+          appendToast: true,
+          dismissible: true,
+          noAutoHide: true
+        })
+        return true
+      }
+      try {
+        await this.readFile(file)
+        return false
+      } catch (error) {
+        this.$bvToast.toast(`Unable to parse file: ${this.file.name}`, {
+          title: 'Error',
+          variant: 'danger',
+          toaster: 'b-toaster-bottom-right',
+          appendToast: true,
+          dismissible: true,
+          noAutoHide: true
+        })
+        return true
+      }
+    },
+    async readFile(file) {
+      return new Promise((resolve, reject) => {
+        const reader = new FileReader()
+        reader.readAsText(file, 'UTF-8')
+        reader.onload = evt => {
+          try {
+            resolve(JSON.parse(evt.target.result))
+          } catch (error) {
+            reject(error)
+          }
+        }
+        reader.onerror = evt => {
+          reject()
+        }
+      })
+    },
+    async fileAdded() {
+      const isInvalidFile = await this.isInvalidFile(this.file)
+      if (!isInvalidFile) {
```

Comment:
Instead of having `if (! isInvalidFile)` you could just have `if (isValidFile)`. Usually simple condition are easier to read